# SARS-CoV-2 Deep Sequencing 

At the University of Wisconsin-Madison, the AIDS Vaccine Research Laboratories have been working together to sequence positive SARS-CoV-2 nasal swab samples from patients that present to the University of Wisconsin Hopsital and Clinic system.

Principal investigators leading these efforts are Drs. Tom Friedrich, Dave O'Connor and Shelby O'Connor. Sequencing efforts, at this point, have been led by [Gage Moreno](https://twitter.com/GageKMoreno) and [myself](https://twitter.com/KATarinambraun). 

We are aiming to sequence as many samples as possible and will write up brief analysis reports as frequently as possible. We will submit consensus sequences to GISAID (which will be fed into [Nextstrain phylogenies](https://nextstrain.org/ncov)) and we will additionally submit cleaned fastq datasets (human reads removed) to SRA to facilitate open datasharing. 

We are additionally tracking these reports on our [LabKey portal](https://openresearch.labkey.com/wiki/ZEST/Ncov/page.view?name=sars-cov-2-sequencing). 

We are sequencing via a number of methods, but moving forward will primarily sequence on an ONT Nanopore GridION and will use the [ARTIC sequencing protocol](https://artic.network/ncov-2019). The earlier reports will include sequencing using a non-specific, metagenomic approach called [SISPA](https://www.protocols.io/view/sequence-independent-single-primer-amplification-o-bb3miqk6). We will include some Illumina MiSeq sequencing as well. 

This repository will be organized by date. I will upload the reports and consensus sequences for the collection of samples we sequenced and processed by date. 
