{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sniffles\n",
    "\n",
    "**This is the first script that should be run to replicate analyses and figure-generation for the Illumina data.**\n",
    "\n",
    "`Sniffles` is a pipeline originally written for the analysis of influenza genomes, but we've modified it here for the analysis of SARS-CoV-2 viruses. It's a configurable pipeline that performs multiple functions to generate variant and consensus level information. It requires minimal dependencies as the pipeline relies on a docker container to host the software.\n",
    "\n",
    "This file contains the config parameters as well as the commands necessary to replicate the analysis we've done for this paper.\n",
    "\n",
    "## Table of Contents\n",
    "* [Requirements](#requirements)\n",
    "* [Installing](#installing)\n",
    "* [Usage](#usage)\n",
    "\n",
    "### Requirements\n",
    "* Linux or MacOS\n",
    "* Python 3.6 or later\n",
    "* Docker\n",
    "\n",
    "### Installing\n",
    "* Make sure you are using the correct version of python\n",
    "* Use git to get the latest sniffles build `git clone https://github.com/k-florek/sniffles.git`\n",
    "* Install the [Docker CE engine](https://docs.docker.com/install/)\n",
    "* Install the required python libraries by pointing the pip installer to the requirements document in the sniffles project `pip3 install -r requirements.txt`\n",
    "\n",
    "### Usage\n",
    "Sniffles uses a configuration file to provide parameters to the pipeline. There is a default configuration file included in the sniffles project. To run the pipeline simple point the `sniffles.py` program at the correct config.yml and directory containing the raw Illumina reads.\n",
    "\n",
    "```\n",
    " #####\n",
    "#     #  #    #  #  ######  ######  #       ######   ####\n",
    "#        ##   #  #  #       #       #       #       #\n",
    " #####   # #  #  #  #####   #####   #       #####    ####\n",
    "      #  #  # #  #  #       #       #       #            #\n",
    "#     #  #   ##  #  #       #       #       #       #    #\n",
    " #####   #    #  #  #       #       ######  ######   ####\n",
    "\n",
    "\n",
    "\n",
    "usage: sniffles.py [-h] [-c config] [-i input] [-o output] [-t threads]\n",
    "\n",
    "Pipeline to examine SNPs from raw illumina reads\n",
    "\n",
    "optional arguments:\n",
    " -h, --help  show this help message and exit\n",
    " -c config   config file\n",
    " -i input    raw reads directory - defaults to working directory\n",
    " -o output   output directory - defaults to working directory\n",
    " -t threads  number of cpus to use for pipeline\n",
    "```\n",
    "\n",
    "Example usage:\n",
    "`./sniffles.py -c config.yml -i ~/my_reads/ -t 8 -o my_results`\n",
    "\n",
    "### Input \n",
    "\n",
    "Raw fastQ files:  \n",
    "`SARSCoV2_passage_MS/data_raw/Illumina_FASTQ`\n",
    "\n",
    "Config file (parameters outlined below):  \n",
    "`SARS-CoV-2/SARSCoV2_passage_MS/data_pipelines/Illumina_pipeline/config.yml`\n",
    "\n",
    "Reference fasta:  \n",
    "`SARSCoV2_passage_MS/data_pipelines/refs_and_GTFs/SARS2_genes.fasta`\n",
    "\n",
    "GTF file:  \n",
    "`SARSCoV2_passage_MS/data_pipelines/refs_and_GTFs/SARS2_genes.gtf`\n",
    "\n",
    "### Output \n",
    "\n",
    "\n",
    "\n",
    "### Config file parameters  \n",
    "\n",
    "The config file \"sample\", \"ref\" and \"GTF\" need to be updated to match sample name before running each sample. \n",
    "\n",
    "```Python\n",
    "#configuration file for holding runtime configurations and parameters\n",
    "\n",
    "# ---Pipeline Execution Parameters---\n",
    "exec:\n",
    "  #name of the output directory\n",
    "  outdir: 'sample'\n",
    "  #name of the logfile\n",
    "  logfile: 'sample_logs.log'\n",
    "  #list of reference sequences used for mapping \n",
    "  #(brackets[] indicate more than one reference. \n",
    "  #If used, inputs must be in a subfolder named for the reference sequence)\n",
    "  referenceSequences: [\"SARS2.fasta\"]\n",
    "  #minimum average depth after mapping\n",
    "  minimumAverageDepth: 100\n",
    "  #minimum percent of reference bases covered\n",
    "  percentRefCovered: 100\n",
    "\n",
    "  # ---replicate variables---\n",
    "  #samples in replicate\n",
    "  replicates: False\n",
    "  #Notation for sample replicates: SampleX_R1.fasta \n",
    "  #eg. if your fastq is H1N1.1_R1, please write \"Sample.1_R1\"\n",
    "  # or if fastq is H1N1Rep1_R1, please write \"SampleRep1_R1\"\n",
    "  # or if fastq is r1_ZW456_R1, please write \"r1_Sample_R1\"\n",
    "  replicateNotation: \"Sample_rep1_R1\"\n",
    "  #variant calling using 'Varscan' or 'RePlow' or 'Compare' (only three options). \n",
    "  #'Compare' will run both and produce an excel file that places the results side by side. \n",
    "  #'Compare' does not work with more than one set of replicates.\n",
    "  #***Compare is not currently supported.***\n",
    "  callSNPs: 'Varscan'\n",
    "  \n",
    "  # ---pipeline functions---\n",
    "  #filter out samples which don't meet minimum coverage threshold\n",
    "  coverageFilter: False\n",
    "  #consensus generation using VarScan\n",
    "  generateConsensus: True\n",
    "  #consensus generation for all samples of given reference using VarScan\n",
    "  generatePopConsensus: True\n",
    "  #map the reads to the consensus sequence using bowtie2\n",
    "  mapToConsensus: False\n",
    "  #annotate SNPs with Louise Moncla's custom annotator\n",
    "  annotateSNPs: True\n",
    "  #use SNPgenie to derive population statistics #we'll do this separately\n",
    "  SNPgenier: False\n",
    "\n",
    "  # ---data cleaning---\n",
    "  #use unpaired reads, or only paired reads\n",
    "  unpaired: False\n",
    "\n",
    "  #coverage depth normalization using seqtk\n",
    "  #normalize coverage across the sam or bam file by randomly downsampling to a set number of reads\n",
    "  normalizeCoverage: False\n",
    "  totalReads: 200000\n",
    "\n",
    "# ---Quality Trimming Parameters---\n",
    "trimmomatic:\n",
    "  #remove adapter sequences from the reads\n",
    "  removeAdapters: False\n",
    "  #adapter file name included in trimmomatic\n",
    "  adaptersFileName: \"Nextera_XT_adapter.fa\"\n",
    "  #paired end reads\n",
    "  paired: True\n",
    "  #minmial read length\n",
    "  minlength: 100\n",
    "  #sliding window size for quality trim\n",
    "  windowSize: 5\n",
    "  #quality threashold for trim in sliding window\n",
    "  qscore: 30\n",
    "\n",
    "# ---Single Nucelotide Polymorhpism Calling Parameters---\n",
    "snpcalling:\n",
    "  #minimum coverage for SNP\n",
    "  minCoverage: 100\n",
    "  #minimum quality for SNP\n",
    "  snpQualityThreshold: 30\n",
    "  #minimum frequency for SNP\n",
    "  snpFrequency: 0.01\n",
    "  #minimum frequency for a consensus SNP in generating the consensus sequence\n",
    "  consensusFrequency: 0.5\n",
    "\n",
    "#Replow-specific settings\n",
    "replow_settings:\n",
    "  #minimum mapping quality\n",
    "  mapquality: 30\n",
    "  #influenza mutation rate\n",
    "  mutrate:  4E-05\n",
    "  \n",
    "#The fundamental question being asked is \"What is the pre-test probability of any given nucleotide being different than the reference sequence?\"\n",
    "#This number is derived in cancer from looking at the number of mutation per Mb in the tumor compared to somatic mutations. You can't really do that in viruses.\n",
    "#The closest analogy would be looking at the number of SNPs present in one ferret after infection with a standard strain. I don't know if people have done this,\n",
    "#Because mutation rate/replication cycle is so much easier to do.\n",
    "#This paper more recently estimated 1.4x10^-5 mutations/site/48hrs in influenza A, which seems like a reasonable estimate for a course of influenza.\n",
    "#source: Nobusawa E, Katsuhiki S. \"Comparison of the Mutation Rates of Human Influenza A and B Viruses\". Journal of Virology Mar 2006, 80 (7) 3675-3678; DOI: 10.1128/JVI.80.7.3675-3678.2006\n",
    "\n",
    "# ---Post-SNP population analysis---\n",
    "postprocessing:\n",
    "  #location of gtf file\n",
    "  gtfFileNames: [\"SARS2.gtf\"]\n",
    "  #minimum frequency of SNPs to be considered by SNPGenie\n",
    "  minSNPfreq: 0.01\n",
    "  #sliding window size for SNPGenie analysis (# of codons)\n",
    "  slidingwindow: 50\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Command\n",
    "\n",
    "For each of the following files: \n",
    "1. p1_vero76\n",
    "2. p1_veroE6\n",
    "3. p1_veroSTAT1KO\n",
    "4. p2a_vero76\n",
    "5. p2b_vero76\n",
    "6. primary_NP_swab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      " #####\n",
      "#     #  #    #  #  ######  ######  #       ######   ####\n",
      "#        ##   #  #  #       #       #       #       #\n",
      " #####   # #  #  #  #####   #####   #       #####    ####\n",
      "      #  #  # #  #  #       #       #       #            #\n",
      "#     #  #   ##  #  #       #       #       #       #    #\n",
      " #####   #    #  #  #       #       ######  ######   ####\n",
      "\t\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "stty: 'standard input': Inappropriate ioctl for device\n",
      "Traceback (most recent call last):\n",
      "  File \"sniffles.py\", line 99, in <module>\n",
      "    sc.procTitle(startRunMessage, cfg)\n",
      "  File \"/Users/katbraun/Documents/research/kat_braun/projects/SARS-CoV-2/SARSCoV2_passage_MS/data_pipelines/Illumina_pipeline/supporting_code/sc.py\", line 30, in procTitle\n",
      "    rows,cols = os.popen('stty size','r').read().split()\n",
      "ValueError: not enough values to unpack (expected 2, got 0)\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b\"\\npython sniffles.py -c config.yml -i '../../data_raw/Illumina_FASTQ' -o 'output'\\n\\n##\\n\"' returned non-zero exit status 1.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-9-9cd0754fb32d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bash'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"\\npython sniffles.py -c config.yml -i '../../data_raw/Illumina_FASTQ' -o 'output'\\n\\n##\\n\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2345\u001b[0m                 \u001b[0mmagic_arg_s\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvar_expand\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstack_depth\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2346\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2347\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2348\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2349\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mnamed_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    140\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    141\u001b[0m                 \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscript\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 142\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshebang\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    144\u001b[0m         \u001b[0;31m# write a basic docstring:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m</Users/katbraun/anaconda3/lib/python3.6/site-packages/decorator.py:decorator-gen-110>\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/IPython/core/magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_error\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mCalledProcessError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    246\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    247\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_run_script\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_close\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b\"\\npython sniffles.py -c config.yml -i '../../data_raw/Illumina_FASTQ' -o 'output'\\n\\n##\\n\"' returned non-zero exit status 1."
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "python sniffles.py -c config.yml -i '../../data_raw/Illumina_FASTQ' -o 'output'\n",
    "\n",
    "##"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
